### PR TITLE
Preparing new major release incl PHP8 support

### PR DESCRIPTION
**Pull request by seehase**
@ericplan
Please have a look

### CODE REVIEW

**Comment by ericplan** [comment]
I'll do it shortly, probably in parts.

File: ZonPHP/charts/chart_styles.php
Comment (ericplan) [inline ZonPHP/charts/chart_styles.php:1]: The lines months, weekdays and shortMonths are not used in the charts, as far as I know. Should be safe to remove this from the chart_styles file.

**Review by ericplan — COMMENTED** [review]
In general OK. Didn't test it yet. One basic thing: Don't know where we should put the date format thing? Do we put it in the xxx_chart.php or in the xxx_overview.php? And with my syntax we keep the original variable $datum. With your syntax you change $datum to a new value. I cannot oversee what the consequences are, but I prefer my solution. $datum is used on several locations. 
$formatter->format(strtotime($datum))
One more suggestion: In the month overview I found out I like the MMM or LLL format better. It is easier to scroll to months because the arrow buttons stay in one place.

Eric

File: ZonPHP/install/par_edit.php [hunk 1]
```diff
@@ -18,8 +18,8 @@
     $_POST['next_mail_threshold'] = $row['Waarde'];
 }
 
-
-$statoe = 1;
+//be careful with this one
+//$statoe = 1;
```
Comment (seehausen) [inline ZonPHP/install/par_edit.php]: For what is that needed anyway?
I just see assignments but no use of the variable

**Comment by ericplan** [comment]
Hi,It bypasses the login. When you have this variable set you get to all the settings without any typingEricVerstuurd vanaf mijn iPadOp 12 mei 2023 om 09:10 heeft Holger Seehausen ***@***.***> het volgende geschreven:﻿
@seehausen commented on this pull request.



In ZonPHP/install/par_edit.php:
> +//be careful with this one
+//$statoe = 1;

For what is that needed anyway?
I just see assignments but no use of the variable

**Comment by seehausen** [comment]
but how does that work? I do not find any usage of the variable.
BTW: I saw there is also a `loeschen.php` isn't is the same as `delete.php` and can also be removed?

**Comment by ericplan** [comment]
Hi,

Maybe it was just a coincidence with $_SESSION['passok’] still active and is it safe to delete.

About par_edit.php: you removed the checkboxes for all the charts, not only the temperature stuff. Mistake or on purpose? Below line 443 there is some orphan function code, which is safe to delete.

I have a look if the tooltip checkbox function is working, I guess not since it comes from the flash charts.

About delete.php and loschen.php (same for actiondelete.php and actionloschen.php) don’t know. Have to look into it.

Regards,

Eric

**Comment by ericplan** [comment]
Hi,

I was using a very old index.php. Now I updated to the new version from the new_major_version page, added four files with js and css but I get an error variable UseDragDrop not found thrown in index_charts.js on line 4

Where do I look? The fault is only on the index page. I’m able to display charts from the menu

Regards,

Eric

**Comment by seehausen** [comment]
recommendation:

- delete all $statoe = 1; statement
- par_edit: chart checkbox removed, because now we only have 7 charts left so no need to configure.
- we can also think about deleting then top/worst chart, does not provide useful information
- delete.php and loschen.php  check , if save delete one
- UseDragDrop: I removed some code because it was used for the Inverter-dropdown  which was also removed, Maybe you have not updated all files, ich I start from scratch, this error do not appear
- debug.php is also throwing some errors, I'll have a look on this
- tooltip checkbox function is working, --> remove (I removed already a lot from the parameter

**Comment by ericplan** [comment]
Hi,

Started from scratch with a fresh download from GitHub in a new directory, I get this warning Warning: Undefined array key "chart_date_format" in /home/deb95965/domains/e2fb.nl/public_html/ZonPHP/charts/chart_styles.php on line 111

Eric

**Comment by ericplan** [comment]
Hi,

It was something old in the database, it is correct now.

Regards,

Eric

**Comment by ericplan** [comment]
Hi,

I will have a look at several things you named. For now I found out the $[SESSION] variables aren’t destroyed when leaving or re-opening site. This is a new issue. So, once logged in, you’ll stay logged in. Also, the login system uses the password from the database. Not safe when you login on a public computer.

Maybe this is a better solution? https://phppot.com/php/php-login-script-with-session/
Or this https://codeshack.io/secure-login-system-php-mysql/

Regards,

Eric

**Comment by seehase** [comment]
I will also have a look, into the session handling, we did not change
anything there

continue working tomorrow

Am Fr., 12. Mai 2023 um 21:32 Uhr schrieb ericplan ***@***.***
>:

> Hi,
>
> I will have a look at several things you named. For now I found out the
> $[SESSION] variables aren’t destroyed when leaving or re-opening site. This
> is a new issue. So, once logged in, you’ll stay logged in. Also, the login
> system uses the password from the database. Not safe when you login on a
> public computer.
>
> Maybe this is a better solution?
> https://phppot.com/php/php-login-script-with-session/
> Or this https://codeshack.io/secure-login-system-php-mysql/
>
> Regards,
>
> Eric
>
>
>

**Comment by ericplan** [comment]
Hi Holger,Same for me. Why I’m bothered at the login procedure is because my database at my provider doesn’t allow strong passwords. Character number is limited to 12, only numbers and normal letters. So I would like a better solution, this is not very secure for me.About the language files, I took a look at the page source and noticed the language file is passed to JavaScript.Does is make sense to split the language files in two separate ones? One used for par_edit, the other for displaying in the charts? But also notice that the language files can be reduced by a great amount because of skipping the color part out of par_edit.php, so maybe there is no need to do it.Regards,EricVerstuurd vanaf mijn iPadOp 12 mei 2023 om 23:02 heeft seehase ***@***.***> het volgende geschreven:﻿
I will also have a look, into the session handling, we did not change
anything there

continue working tomorrow

Am Fr., 12. Mai 2023 um 21:32 Uhr schrieb ericplan ***@***.***
>:

> Hi,
>
> I will have a look at several things you named. For now I found out the
> $[SESSION] variables aren’t destroyed when leaving or re-opening site. This
> is a new issue. So, once logged in, you’ll stay logged in. Also, the login
> system uses the password from the database. Not safe when you login on a
> public computer.
>
> Maybe this is a better solution?
> https://phppot.com/php/php-login-script-with-session/
> Or this https://codeshack.io/secure-login-system-php-mysql/
>
> Regards,
>
> Eric
>
>
>

**Comment by seehase** [comment]
@ericplan can you check the https://github.com/seehase/ZonPHP/issues if they are still valid?
if not we can close them

**Review by seehase — COMMENTED** [review]
Do we want to keep the top31.php and chart?

**Comment by ericplan** [comment]
Yes, the top31.chart is nice, (but only in May, June) The top31_overview was old and not updated.

**Comment by ericplan** [comment]
Maybe we could keep the top 31 chart, but remove it from the index page?

**Comment by seehase** [comment]
> Maybe we could keep the top 31 chart, but remove it from the index page?

That can be done easy: 
just remove/comment line
`        addBest();`
in index_chart.js line 70

**Comment by ericplan** [comment]
Catch22 ?
Busy with some cosmetic and some more fundamental changes on the installation side. I would like to create a new database with another prefix, not tgeg. But from the script, this seems impossible. This should be selectable and it makes it also possible to check if it works correct without risking the current installation.

**Comment by ericplan** [comment]
What is /install/opstart_installatie.php doing. I only get errors when running it. Looks obsolete and safe to delete.

**Comment by ericplan** [comment]
Have a look. As far as I can see update.php and actionupdate.php is all about earnings. Can be deleted?

**Comment by ericplan** [comment]
Same for delete and actiondelete?!

**Comment by seehase** [comment]
Sensor data table can be removed, as well as the other marked tables @ericplan

**Comment by seehase** [comment]
Eric and I did a lot rework cleanup and improvments
Changes are squashed and merged into master

**Pull request closed without merging**